user_guide_src/source/testing/overview/018.php

```diff
@@ -2,26 +2,19 @@
 
 use CodeIgniter\CLI\CLI;
 use CodeIgniter\Test\CIUnitTestCase;
-use CodeIgniter\Test\Filters\CITestStreamFilter;
+use CodeIgniter\Test\StreamFilterTrait;
 
 final class SomeTest extends CIUnitTestCase
 {
-    protected function setUp(): void
-    {
-        CITestStreamFilter::$buffer = '';
-        $this->stream_filter        = stream_filter_append(STDOUT, 'CITestStreamFilter');
-    }
-
-    protected function tearDown(): void
-    {
-        stream_filter_remove($this->stream_filter);
-    }
+    use StreamFilterTrait;
 
     public function testSomeOutput(): void
     {
+        $this->resetStreamFilterBuffer();
+
         CLI::write('first.');
 
         $expected = "first.\n";
```

Comment:
The assertion failed.

```suggestion
        $expected = "\nfirst.\n";
```